pretty_gpx/rendering_modes/city/city_poster_image_cache.py

```diff
@@ -0,0 +1,263 @@
+#!/usr/bin/python3
+"""Poster Image Cache."""
+from dataclasses import dataclass
+
+from matplotlib.axes import Axes
+from matplotlib.figure import Figure
+
+from pretty_gpx.common.data.overpass_request import OverpassQuery
+from pretty_gpx.common.drawing.base_drawing_figure import BaseDrawingFigure
+from pretty_gpx.common.drawing.drawing_data import BaseDrawingData
+from pretty_gpx.common.drawing.drawing_data import LineCollectionData
+from pretty_gpx.common.drawing.drawing_data import PlotData
+from pretty_gpx.common.drawing.drawing_data import PolyFillData
+from pretty_gpx.common.drawing.drawing_data import PolygonCollectionData
+from pretty_gpx.common.drawing.drawing_data import ScatterData
+from pretty_gpx.common.drawing.drawing_data import TextData
+from pretty_gpx.common.gpx.gpx_bounds import GpxBounds
+from pretty_gpx.common.layout.paper_size import PaperSize
+from pretty_gpx.common.utils.logger import logger
+from pretty_gpx.common.utils.profile import profile
+from pretty_gpx.common.utils.utils import format_timedelta
+from pretty_gpx.common.utils.utils import mm_to_point
+from pretty_gpx.rendering_modes.city.city_drawing_figure import CityDrawingFigure
+from pretty_gpx.rendering_modes.city.city_vertical_layout import CityVerticalLayout
+from pretty_gpx.rendering_modes.city.data.augmented_gpx_data import CityAugmentedGpxData
+from pretty_gpx.rendering_modes.city.data.forests import prepare_download_city_forests
+from pretty_gpx.rendering_modes.city.data.forests import process_city_forests
+from pretty_gpx.rendering_modes.city.data.rivers import prepare_download_city_rivers
+from pretty_gpx.rendering_modes.city.data.rivers import process_city_rivers
+from pretty_gpx.rendering_modes.city.data.roads import prepare_download_city_roads
+from pretty_gpx.rendering_modes.city.data.roads import process_city_roads
+from pretty_gpx.rendering_modes.city.drawing.city_drawing_params import CityDrawingStyleParams
+from pretty_gpx.rendering_modes.city.drawing.city_drawing_params import CityLinewidthParams
+from pretty_gpx.rendering_modes.city.drawing.theme_colors import ThemeColors
+
+W_DISPLAY_PIX = 800  # Display width of the preview (in pix)
+
+WORKING_DPI = 50  # DPI of the poster's preview
+HIGH_RES_DPI = 400  # DPI of the final poster
+
+
+@dataclass
+class CityPosterImageCaches:
+    """Low and High resolution CityPosterImageCache."""
+    low_res: 'CityPosterImageCache'
+    high_res: 'CityPosterImageCache'
```

Comment:
no need to have high and low res since there's no rasterized component to drawn, everything is vectorized in the city mode ! :)

Reply:
You are right, I have edited the code.